core/discovery/discovery.go

```diff
@@ -44,6 +45,45 @@ const (
 
 const logPrefix = "[discovery] "
 
+// Discovery structure holds discovery service state
+type Discovery struct {
+	identityRegistry            identity_registry.IdentityRegistry
+	ownIdentity                 identity.Identity
+	identityRegistration        identity_registry.RegistrationDataProvider
+	proposalRegistry            ProposalRegistry
+	signerCreate                identity.SignerFactory
+	signer                      identity.Signer
+	proposal                    market.ServiceProposal
+	statusChan                  chan Status
+	status                      Status
+	proposalAnnouncementStopped *sync.WaitGroup
+	unsubscribe                 func()
+	stop                        func()
```

Comment:
It looks like `stop` and `unsubscribe` are doing the same things.

Reply:
Existing code, not my scope